### PR TITLE
Fix NoConsensus Mode. Refactor tests. Test NoConsensus mode.

### PR DESCRIPTION
**Pull request by silasdavis**
No consensus mode was not resetting the checker which meant CallTxSim was not working.

This PR fixes this and also heavily refactors tests to avoid the use of main_test which leads to poor encapsulation and stops us from running tests over different kernel configurations.

Subtests can be used to achieve the same effect and also lead to easier parallelism. Reworked the integration test helpers to facilitate this.

### CODE REVIEW

File: core/config.go [hunk 1]
```diff
@@ -119,23 +107,37 @@ func LoadKernelFromConfig(conf *config.BurrowConfig) (*Kernel, error) {
 		return nil, fmt.Errorf("could not create initial kernel: %v", err)
 	}
 
-	if err = kern.LoadLoggerFromConfig(conf.Logging); err != nil {
+	err = kern.LoadLoggerFromConfig(conf.Logging)
```
Comment (gregdhill) [inline core/config.go]: Why are you reverting these? I personally think it's a lot cleaner.

File: tests/chain/burrow.toml [hunk 2]
```diff
@@ -7,6 +7,7 @@
   TendermintRoot = ".burrow"
```
Comment (gregdhill) [inline tests/chain/burrow.toml]: This should be `BurrowDir` now

```suggestion
   BurrowDir = ".burrow"
```

File: core/kernel.go [hunk 3]
```diff
@@ -228,38 +243,26 @@ func (kern *Kernel) LoadTransactor() (err error) {
...
+		return nil
 	}
+
+	// Use the Tendermint blockstore
+	kern.Blockchain.SetBlockStore(bcm.NewBlockStore(nodeView.BlockStore()))
+	// Provide execution accounts against checker state so that we can assign sequence numbers
+	accounts := execution.NewAccounts(kern.checker, kern.keyClient, AccountsRingMutexCount)
+	// Pass transactions to Tendermint's CheckTx function for broadcast and consensus
+	checkTx := kern.Node.MempoolReactor().Mempool.CheckTx
+	kern.Transactor = execution.NewTransactor(kern.Blockchain, kern.Emitter, accounts, checkTx, kern.txCodec,
```
Comment (gregdhill) [inline core/kernel.go]: If you want to simplify this even further I guess you only need to set `checkTx` and possibly `kern.Blockchain` if in consensus mode.

File: consensus/abci/app.go [hunk 4]
```diff
@@ -192,90 +181,52 @@ func (app *App) checkValidatorMatches(ours validator.Reader, v abciTypes.Validat
...
-		Code: codes.TxExecutionSuccessCode,
-		Log:  logf("Execution success - TxExecution in data"),
-		Data: bs,
+	if checkTx.Code != codes.TxExecutionSuccessCode {
+		logger.InfoMsg("Execution error",
+			"code", checkTx.Code,
+			"log", checkTx.Log)
 	}
 
+	logger.InfoMsg("Execution success")
```
Comment (seanyoung) [inline consensus/abci/app.go]: Why are logging "Execution succcess" just after logging "Execution error"?

File: consensus/abci/app.go [hunk 5]
```diff
@@ -192,90 +181,52 @@ func (app *App) checkValidatorMatches(ours validator.Reader, v abciTypes.Validat
...
+
+	logger := WithTags(app.logger, checkTx.Tags)
+
+	if checkTx.Code != codes.TxExecutionSuccessCode {
+		logger.InfoMsg("Execution error",
+			"code", checkTx.Code,
+			"log", checkTx.Log)
+	}
+
+	logger.InfoMsg("Execution success")
```
Comment (seanyoung) [inline consensus/abci/app.go]: same here

File: CHANGELOG.md [hunk 6]
```diff
@@ -2,6 +2,23 @@
...
 
+## [0.25.0] - 2019-04-05
+### Changed
+- [Tendermint] Upgraded to 0.31.2
+- [IAVL] upgraded to 0.12.2
+- [Config] Tendermint.TimeoutFactor moved to Execution.TimeoutFactor (and reused for NoConsensus mode)
+- [Kernel] Refactored and various exported methods changed
+
+### Added
+- [CLI] Burrow deploy can now run jobs in parallel according to their dependency structure
```
Comment (seanyoung) [inline CHANGELOG.md]: That's not in place yet. All there is that burrow deploy can execute multiple burrow deploy files, and execute them in parallell.

**Pull request merged**